test/parse.jl

```diff
@@ -771,19 +771,26 @@ macro iter()
 end
 end
 let ex = expand(M16096, :(@iter))
-    @test isa(ex, Expr) && ex.head === :body
+    @test isa(ex, Expr) && ex.head === :thunk
 end
 let ex = expand(Main, :($M16096.@iter))
-    @test isa(ex, Expr) && ex.head === :body
+    @test isa(ex, Expr) && ex.head === :thunk
 end
 let ex = expand(@__MODULE__, :(@M16096.iter))
-    @test isa(ex, Expr) && ex.head === :body
+    @test isa(ex, Expr) && ex.head === :thunk
     @test !isdefined(M16096, :foo16096)
-    @test eval(@__MODULE__, ex) === nothing
+    local_foo16096 = eval(@__MODULE__, ex)
+    @test local_foo16096(2.0) == 1
     @test !@isdefined foo16096
-    @test isdefined(M16096, :foo16096)
+    @test !@isdefined it
+    @test !isdefined(M16096, :foo16096)
+    @test !isdefined(M16096, :it)
+    @test_broken typeof(local_foo16096).name.module === M16096
+    @test_broken typeof(local_foo16096).name.mt.module === M16096
+    @test_broken getfield(M16096, typeof(local_foo16096).name.mt.name) === local_foo16096
+    @test_broken !isdefined(@__MODULE__, typeof(local_foo16096).name.mt.name)
```

Comment:
> as defining a method in the caller's module

Also, fwiw, it's best to be precise in terminology, since this test has nothing to do with `Method`, but instead is testing for the properties of the function's type. Macroexpand (and expand in general) would require severe contortions (spliced in calls to `eval`) to affect that property. This instead is testing specifically for whether the macro added a new global bindings (and where).